features/networking/multus.feature

```diff
@@ -1056,3 +1056,115 @@ Feature: Multus-CNI related scenarios
+    Given I have a project
+    When I run oc create as admin over "https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/networking/multus-cni/NetworkAttachmentDefinitions/ipvlan-host-local.yaml" replacing paths:
+      | ["metadata"]["name"]      | myipvlan76 											      								            |
+      | ["metadata"]["namespace"] | <%= project.name %> 															                            |    
+      | ["spec"] ["config"]       | '{ "cniVersion": "0.3.1", "name": "myipvlan76", "type": "ipvlan", "master": "<%= cb.default_interface %>", "ipam": { "type": "host-local", "subnet": "22.2.2.0/24" } }' |
+    Then the step should succeed
+
+    #Creating various pods and making sure their mac matches to default inf and they get unique IPs assigned
+    #Creating pod1 absorbing above net-attach-def
+    When I run oc create over "https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/networking/multus-cni/Pods/generic_multus_pod.yaml" replacing paths:
+      | ["metadata"]["name"]                                       | pod1                    |
+      | ["metadata"]["annotations"]["k8s.v1.cni.cncf.io/networks"] | myipvlan76              |
+      | ["spec"]["nodeName"]                                       | <%= cb.nodes[0].name %> |
+    Then the step should succeed
+    And the pod named "pod1" becomes ready
+    When I execute on the pod:
+      | /usr/sbin/ifconfig | net1 |
+    Then the step should succeed
+    And evaluation of `@result[:response].match(/\d{1,3}\.\d{1,3}.\d{1,3}.\d{1,3}/)[0]` is stored in the :pod1_net1_ip clipboard
+    And evaluation of `@result[:response].match(/\w{2}:\w{2}:\w{2}:\w{2}:\w{2}:\w{2}/)[0]` is stored in the :pod1_net1_mac clipboard
+    And the expression should be true> cb.pod1_net1_mac==cb.default_interface_mac
+    
+    #Creating pod2 absorbing above net-attach-def
+    When I run oc create over "https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/networking/multus-cni/Pods/generic_multus_pod.yaml" replacing paths:
+      | ["metadata"]["name"]                                       | pod2                    |
+      | ["metadata"]["annotations"]["k8s.v1.cni.cncf.io/networks"] | myipvlan76              |
+      | ["spec"]["nodeName"]                                       | <%= cb.nodes[0].name %> |
+    Then the step should succeed
+    And the pod named "pod2" becomes ready
+    When I execute on the pod:
+      | /usr/sbin/ifconfig | net1 |
+    Then the step should succeed
+    And evaluation of `@result[:response].match(/\d{1,3}\.\d{1,3}.\d{1,3}.\d{1,3}/)[0]` is stored in the :pod2_net1_ip clipboard
+    And evaluation of `@result[:response].match(/\w{2}:\w{2}:\w{2}:\w{2}:\w{2}:\w{2}/)[0]` is stored in the :pod2_net1_mac clipboard
+    And the expression should be true> cb.pod2_net1_mac==cb.default_interface_mac
+    And the expression should be true> !(cb.pod2_net1_ip==cb.pod1_net1_ip)
+    
+    #Creating pod3 absorbing above net-attach-def
```

Comment:
@anuragthehatter When create the pods your steps is using <%= cb.nodes[0].name %> and make sure those pods are created in the nodes[0], then I think checking two pods mac will good enough, you do not need check pod3 and pod4. The origin test case dose not use <%= cb.nodes[0].name %> and pods are created cross all three nodes, then  4 pods are needed for testing purpose. You can delete line 1139-1170.
Then /LGTM

Reply:
Yep i agree, being on same node we don't need so many pods as two will be enough to check master macs. Thanks. Fixed and re-tested http://pastebin.test.redhat.com/843986

Reply:
/LGTM